kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java

```diff
@@ -330,4 +336,51 @@ private static Literal literalForPartitionValue(DataType dataType, String partit
 
         throw new UnsupportedOperationException("Unsupported partition column: " + dataType);
     }
+
+    /**
+     * Serialize the given partition value to a string according to the Delta protocol <a
+     * href="https://github.com/delta-io/delta/blob/master/PROTOCOL.md
+     * #partition-value-serialization">partition value serialization rules</a>.
+     *
+     * @param literal Literal representing the partition value of specific datatype.
+     * @return Serialized string representation of the partition value.
+     */
+    protected static String serializePartitionValue(Literal literal) {
+        Object value = literal.getValue();
+        if (value == null) {
+            return null;
+        }
+        DataType dataType = literal.getDataType();
+        if (dataType instanceof ByteType
+                || dataType instanceof ShortType
+                || dataType instanceof IntegerType
+                || dataType instanceof LongType
+                || dataType instanceof FloatType
+                || dataType instanceof DoubleType
+                || dataType instanceof BooleanType) {
+            return String.valueOf(value);
+        } else if (dataType instanceof StringType) {
+            return (String) value;
+        } else if (dataType instanceof DateType) {
+            int daysSinceEpochUTC = (int) value;
+            return LocalDate.ofEpochDay(daysSinceEpochUTC).toString();
+        } else if (dataType instanceof TimestampType || dataType instanceof TimestampNTZType) {
+            long microsSinceEpochUTC = (long) value;
+            long seconds = microsSinceEpochUTC / 1_000_000;
+            long microsOfSecond = (microsSinceEpochUTC % 1_000_000);
```

Comment:
can this be an int? Since it is <= 1_000_000

Reply:
I only ask because: this is a `long`, and `nanosOfSecond` is an `int`, and we make `nanosOfSecond` be 100x larger than `microsOfSecond`. so it's a bit weid that an int is 1000x larger than a long

Reply:
yep, this can be an int.